### ISSUE TITLE
[22] Implement Elements#getEnumConstantBody()

### ISSUE THREAD

**Issue by jarthana** [eclipse-jdt/eclipse.jdt.core#2135]
This is a new API added in Java 22 and the default implementation throws `UnsupportedOperationException`.

**Comment by jarthana** [comment]
The PR contains just the testcase as of now. I wonder how Javac manages to do this. In ECJ, the annotation elements only get the bindings that don't often get access to their declaration.

Besides, as the tests demonstrate, Javac also fails in binary mode. I.e. the getEnumConstantBody returns null for binary elements.

**Comment by jarthana** [comment]
I happened to revisit this. So, it appears we need to concoct a TypeDeclaration from a given FieldDeclaration with initializer that represents an Enum Constant. I see that DOM has a similar requirement and it appears to create a DOM equivalent on the fly.

Copying the experts - @srikanth-sankaran and @stephan-herrmann 

Is there anyway we can get a TypeDeclaration from the given FieldDeclaration#initializer, which is usually an AllocationExpression?

**Comment by srikanth-sankaran** [comment]
> I happened to revisit this. So, it appears we need to concoct a TypeDeclaration from a given FieldDeclaration with initializer that represents an Enum Constant. I see that DOM has a similar requirement and it appears to create a DOM equivalent on the fly.
> 
> Copying the experts - [@srikanth-sankaran](https://github.com/srikanth-sankaran) and [@stephan-herrmann](https://github.com/stephan-herrmann)
> 
> Is there anyway we can get a TypeDeclaration from the given FieldDeclaration#initializer, which is usually an AllocationExpression?

Yes, see `org.eclipse.jdt.internal.compiler.ast.QualifiedAllocationExpression.anonymousType` and the javadoc above `org.eclipse.jdt.internal.compiler.ast.QualifiedAllocationExpression` - Does this cover your need ?

**Comment by jarthana** [comment]
> > Is there anyway we can get a TypeDeclaration from the given FieldDeclaration#initializer, which is usually an AllocationExpression?
> 
> Yes, see `org.eclipse.jdt.internal.compiler.ast.QualifiedAllocationExpression.anonymousType` and the javadoc above `org.eclipse.jdt.internal.compiler.ast.QualifiedAllocationExpression` - Does this cover your need ?

Thanks, that almost helped! I do see the anonymousType, but unfortunately it's not yet resolved and therefore no binding available since annotations are processed before we had a chance to do CUD#resolve().

**Comment by stephan-herrmann** [comment]
Looking at `QAE.anonymousType` sounds badly like investigating all local and anonymous types ahead of time (which would be a big issue, because such types need a lot of context to be known). But if it's only about constants of non-local enum types, then perhaps we can extract some code around `ClassScope.buildAnonymousTypeBinding(SourceTypeBinding, ReferenceBinding)` to create the binding when needed - and make sure the same is not executed twice :)

**Comment by jarthana** [comment]
> Looking at `QAE.anonymousType` sounds badly like investigating all local and anonymous types ahead of time (which would be a big issue, because such types need a lot of context to be known). But if it's only about constants of non-local enum types, then perhaps we can extract some code around `ClassScope.buildAnonymousTypeBinding(SourceTypeBinding, ReferenceBinding)` to create the binding when needed - and make sure the same is not executed twice :)

Stephan, I have captured your recommendation in https://github.com/eclipse-jdt/eclipse.jdt.core/pull/2139. I don't want to trouble you, but when you have time, please glance at the changes. The changes are mostly in apt, but some refactoring had to be done on compiler also.
